spec/ui_spec.rb

```diff
@@ -1,131 +1,173 @@
 require 'ui'
 
-class TestDisplayForUI
-  attr_reader :displayed
+describe 'UI' do
+  describe '#get_move' do
+    context 'when user input is requested' do
+      it 'prompts the player and returns their move' do
+        class PromptsPlayerFor
+          def initialize(move:)
+            @move = move
+          end
+
+          def call
+            @move
+          end
+        end
 
-  def initialize
-    @displayed = []
-  end
+        class DoesNotDisplayOutput
+        end
 
-  def present(board)
-    displayed << board
-  end
+        prompter = PromptsPlayerFor.new(move: '1')
+        no_display_needed = DoesNotDisplayOutput.new
 
-  def message(message)
-    displayed << message
-  end
-end
-
-class TestPrompterForUI
-  attr_reader :called
+        ui = UI.new(display: no_display_needed, prompter:)
```

Comment:
**praise**: 🤩 

I appreciate that you are using intention-revealing names to call out that this collaborator is not needed for this example!